src/app/screens/Calculator/components/Button/index.js

```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import {View, Text, TouchableHighlight} from 'react-native';
+import styles from './styles.js';
+
+const Button = ({button}) => {
+  return (
+    <TouchableHighlight
+      activeOpacity={0.45}
+      underlayColor="#4bcaf9"
+      onPress={button.action}
+      style={[styles.boxButton, button.label === '0' && styles.specialButton]}>
```

Comment:
This is not the right way to create a variant to button style. What if I want to do a "specialButton" but with the label "ESPECIAL"?.

If you need to make variants, then use a variant prop or if there will only be 2, a flag.

```javascript
const Button = ({label, onPress, style, variant}) => {
...
style={[styles.boxButton, styles[variant]]}>
...
}
```

or 

```javascript
const Button = ({label, onPress, style, special}) => {
...
style={[styles.boxButton, special && styles.specialButton]}>
...
}
```

First approach is way better though, because it's clearer and more flexible.

Reply:
I understand, so I will be implementing it that way.